scim2-sdk-common/src/main/java/com/unboundid/scim2/common/utils/Parser.java

```diff
@@ -161,6 +177,65 @@ public static Filter parseFilter(final String filterString)
     return readFilter(new StringReader(filterString.trim()), false);
   }
 
+  /**
+   * Add parser options, in order to modify parser behavior within the current
+   * thread.
+   *
+   * <p>NOTE: SCIM server implementations are not guaranteed to support a given option.</p>
+   * <p>NOTE: These should be reset when no longer needed.</p>
+   *
+   * <pre>
+   *   Set&lt;ParserOption&gt; priorOptions = null;
+   *   try
+   *   {
+   *     priorOptions = Parser.addOptions(ParserOption.ALLOW_SEMICOLONS_IN_ATTRIBUTE_NAMES);
```

Comment:
I'd move this out of the try block so you don't have to check for null in the finally block. Having things like this outside of the try block is a more typical pattern, like for grabbing a lock and releasing it.

Reply:
OK
```
 Set<ParserOption> priorOptions =
     Parser.addOptions(ParserOption.ALLOW_SEMICOLONS_IN_ATTRIBUTE_NAMES);
 try
 {
   performWhateverProcessing();
 }
 finally
 {
   Parser.setOptions(priorOptions);
 }
 ```